[docs] Rewrite introduction (#2849)
apollographql/federation: docs/source/index.mdx

---
title: Introduction to Apollo Federation
subtitle: Learn how to combine your GraphQL APIs into a unified supergraph
description: Learn how Apollo Federation can help you declaratively combine your GraphQL APIs into a unified, federated supergraph using microservices architecture
---

## What is Apollo Federation?

**Apollo Federation** lets you combine multiple GraphQL APIs into a unified graph with a single entry point called the **router**.
Clients make requests to the router, which intelligently distributes them across your APIs and returns a unified response.
This architecture enables clients to interact with multiple services with a single request.

```mermaid
graph LR;
  clients(Clients);
  router([Router]);
  serviceA[GraphQL API <br/>A];
  serviceB[GraphQL API <br/>B];
  serviceC[GraphQL API <br/>C];
  router --- serviceA & serviceB & serviceC;
  clients -.- router;
  class clients secondary;
```

For performance and security reasons, clients should only query the router, and only the router should query the constituent APIs.
For clients, the request and response cycle of querying the router looks the same as querying any GraphQL server.

> No client-side configuration is required.

Author: Glad you chimed in on this. I went back and forth on this, moving it from the intro to the "Next steps/terminology" section. I'm actually thinking it might fit well under a new Benefits H3 called "Preserve client simplicity" or something like that. LMK if you had other thoughts about where this belongs.

  serviceA[Subgraph A];
  serviceB[Subgraph B];
  serviceC[Subgraph C];
  router --- serviceA & serviceB & serviceC;
  end;
  clients -.- router;
  class clients secondary;
```

The router is a separate service that exposes a GraphQL endpoint to external clients. It serves as the public access point for your supergraph. _Only_ the router queries subgraphs—clients never do so directly.

The router receives incoming GraphQL operations from clients and intelligently routes them across your subgraphs. The router then receives subgraph responses and returns a single response to the client. **For clients, the request and response cycle looks exactly the same as querying any other GraphQL server.** No client-side configuration is required.

<Tip>

Learn more about [your router options](./building-supergraphs/router).

</Tip>

## Benefits of federation

As a GraphQL-native microservices architecture, Apollo Federation draws on both paradigms' benefits.
Federated architecture imparts some unique additional benefits by combining GraphQL and microservices frameworks.
Continue reading to learn more.

<Tip>

- Learn more about the [considerations and benefits of GraphQL](/intro/benefits/).

> - Learn more about the [considerations and benefits of microservices architecture](https://aws.amazon.com/compare/the-difference-between-monolithic-and-microservices-architecture/).

Author: Also considered https://www.atlassian.com/microservices/microservices-architecture/microservices-vs-monolith, but AWS seemed like it would appeal more to our audience.

---
title: Introduction to Apollo Federation
subtitle: Learn how to combine your GraphQL APIs into a unified supergraph

> description: Learn how Apollo Federation can help you combine your GraphQL APIs into a unified, federated supergraph

Author: This serves as an SEO meta description. There may be a few more keywords to add here.

---
title: Introduction to Apollo Federation
subtitle: Learn how to combine your GraphQL APIs into a unified supergraph
description: Learn how Apollo Federation can help you declaratively combine your GraphQL APIs into a unified, federated supergraph using a microservices architecture
---

## What is Apollo Federation?

**Apollo Federation** lets you declaratively combine multiple GraphQL APIs into a unified graph. This enables clients to interact with multiple APIs through a single request.

> A client makes a request to the single entry point of the federated graph called the **router**. The router intelligently orchestrates and distributes the request across your APIs and returns a unified response. For a client, the request and response cycle of querying the router looks the same as querying any GraphQL server.

Reviewer: ```suggestion
A client makes a request to the single entry point of the federated graph called the **router**. The router intelligently orchestrates and distributes the request across your APIs and returns a combined response. For a client, the request and response cycle of querying the router looks the same as querying any GraphQL server.
```

I just feel like I read "unified" too many times in a row.

```mermaid
graph LR;
  clients(Clients);
  router([Router]);
  serviceA[GraphQL API <br/>A];
  serviceB[GraphQL API <br/>B];
  serviceC[GraphQL API <br/>C];
  router --- serviceA & serviceB & serviceC;
  clients -.- router;
  class clients secondary;
```

## Benefits of federation

> ### Microservices architecture

Author: Yeah, I think the GraphQL is super inherent to Apollo so we don't need to call it out.

  end;

  subgraph " ";
    databaseB[(Flight<br/>data)];
    serviceB[Flights API];
    databaseB --- serviceB;
  end;

  subgraph " ";
    databaseC[(Hotel<br/>data)];
    serviceC[Hotels API];
    databaseC --- serviceC;
  end;

  clients(Clients);
  serviceA & serviceB & serviceC -.- clients;
  class clients secondary;
```

> But if you expose multiple GraphQL APIs to clients, a client might need to communicate with _multiple_ APIs to fetch all the data it needs. With a single federated graph, you preserve a powerful advantage of GraphQL over traditional REST APIs: the ability to fetch all the data you need in a single request.

Reviewer 2: ```suggestion
By exposing multiple GraphQL APIs to a client, it might need to communicate with _multiple_ APIs to fetch its data. With a single federated graph, you unlock a powerful advantage of GraphQL over traditional REST APIs: the ability to fetch all the data you need in a single request.
```
